frontend/src/metabase/browse/components/TableBrowser/TableBrowser.unit.spec.js

```diff
@@ -44,24 +49,35 @@ describe("TableBrowser", () => {
   });
 
   it("should render tables with a sync error", () => {
+    const database = getDatabase({
+      initial_sync_status: "incomplete",
+    });
+
     const tables = [
       getTable({ id: 1, name: "Orders", initial_sync_status: "aborted" }),
     ];
 
     render(
       <TableBrowser
+        database={database}
         tables={tables}
         getTableUrl={getTableUrl}
         xraysEnabled={true}
       />,
     );
 
     expect(screen.getByText("Orders")).toBeInTheDocument();
-    expect(screen.queryByLabelText("bolt icon")).not.toBeInTheDocument();
-    expect(screen.getByTestId("loading-spinner")).toBeInTheDocument();
+    expect(screen.getByLabelText("bolt icon")).toBeInTheDocument();
+    expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument();
```

Comment:
If there was an error during the initial sync, `initial_sync_status` won't ever change. We shouldn't continue showing loaders for such tables.

Reply:
Some tables are marked as "cruft" and not synced. Their initial-sync-status will always be "incomplete" and this does not indicate an error. Adding the application db to a clean instance:

```sql
clean=# select id, name, active, visibility_type, initial_sync_status from metabase_table where initial_sync_status = 'incomplete';
 id |                 name                 | active | visibility_type | initial_sync_status
----+--------------------------------------+--------+-----------------+---------------------
 17 | collection_permission_graph_revision | t      | cruft           | incomplete
 38 | query_cache                          | t      | cruft           | incomplete
 49 | databasechangeloglock                | t      | cruft           | incomplete
 53 | databasechangelog                    | t      | cruft           | incomplete
 75 | permissions_revision                 | t      | cruft           | incomplete
 62 | application_permissions_revision     | t      | cruft           | incomplete
(6 rows)

```

This almost feels like a logic issue rather than a code. The sync status for a table is set when we sync the foreign keys for a table. And we only sync the foreign keys for tables that are returned from `sync-util/db->sync-tables` which explicitly looks for `visibility_type = nil`.

Reply:
Yeah, got it now. That's why we don't show spinners in this case anymore.